tests/docker_cluster.py

```diff
@@ -334,67 +336,80 @@ def exec_cmd_on_host(self, host, cmd, raise_error=True, tty=False):
         return output
 
     @staticmethod
-    def start_presto_cluster(install_func):
-        if DockerCluster._check_for_presto_images():
-            presto_cluster = DockerCluster('master',
-                                           ['slave1', 'slave2', 'slave3'],
-                                           DEFAULT_LOCAL_MOUNT_POINT,
-                                           DEFAULT_DOCKER_MOUNT_POINT)
-            presto_cluster.start_containers(INSTALLED_PRESTO_TEST_MASTER_IMAGE,
-                                            INSTALLED_PRESTO_TEST_SLAVE_IMAGE)
-            # replace the node uuid on all the workers
-            for worker in presto_cluster.slaves:
-                presto_cluster.run_script_on_host(
-                    'sed -i /node.id/d /etc/presto/node.properties; '
-                    'uuid=$(uuidgen); '
-                    'echo node.id=$uuid >> /etc/presto/node.properties',
-                    worker
-                )
-            return presto_cluster
-        else:
-            presto_cluster = DockerCluster.start_base_cluster()
-            install_func(presto_cluster)
-            presto_cluster.client.commit(
-                presto_cluster.master,
-                INSTALLED_PRESTO_TEST_MASTER_IMAGE
+    def _get_master_image_name(cluster_type):
+        return os.path.join(DockerCluster.IMAGE_NAME_BASE,
+                            '%s_master' % (cluster_type))
+
+    @staticmethod
+    def _get_slave_image_name(cluster_type):
+        return os.path.join(DockerCluster.IMAGE_NAME_BASE,
+                            '%s_slave' % (cluster_type))
+
+    @staticmethod
+    def start_bare_cluster():
+        dc = DockerCluster
+        master_name = dc._get_master_image_name(dc.BARE_CLUSTER_TYPE)
+        slave_name = dc._get_slave_image_name(dc.BARE_CLUSTER_TYPE)
+        centos_cluster = DockerCluster('master',
+                                       ['slave1', 'slave2', 'slave3'],
+                                       DEFAULT_LOCAL_MOUNT_POINT,
+                                       DEFAULT_DOCKER_MOUNT_POINT)
+
+        if not dc._check_for_images(master_name, slave_name):
+            centos_cluster.create_image(
+                BASE_TD_DOCKERFILE_DIR,
+                master_name,
+                BASE_IMAGE_NAME,
+                BASE_IMAGE_TAG
             )
-            presto_cluster.client.commit(
-                presto_cluster.slaves[0],
-                INSTALLED_PRESTO_TEST_SLAVE_IMAGE
+
+            centos_cluster.create_image(
+                BASE_TD_DOCKERFILE_DIR,
+                slave_name,
+                BASE_IMAGE_NAME,
+                BASE_IMAGE_TAG
             )
-            return presto_cluster
+
+        centos_cluster.start_containers(master_name, slave_name)
+
+        return centos_cluster
 
     @staticmethod
-    def start_base_cluster():
+    def start_existing_images(cluster_type):
+        DC = DockerCluster
```

Comment:
should be lower case. I'm not sure I like this abbreviation; it doesn't fit in with the presto style.